pkg/sql/show_create_external_connection.go

```diff
@@ -31,15 +31,19 @@ var showCreateExternalConnectionColumns = colinfo.ResultColumns{
 	{Name: "create_statement", Typ: types.String},
 }
 
-func loadExternalConnections(
-	params runParams, n *tree.ShowCreateExternalConnections,
-) ([]externalconn.ExternalConnection, error) {
-	var connections []externalconn.ExternalConnection
-	var rows []tree.Datums
+const (
+	createConnNameIdx = iota
+	createConnStmtIdx
+)
 
-	if n.ConnectionLabel != nil {
+func getConnectionNames(
+	params runParams, connectionLabel tree.Expr, checkUsagePrivilege bool,
+) ([]string, error) {
+	var names []string
```

Comment:
If you wanted you could initialise this with the size of `rows` (`make([]string, 0, len(rows)`) below, closer to where it is used, under the assumption that the common case is that the user has permission.